### ISSUE TITLE
Hot lib reload, unable to find __rust_alloc.

### ISSUE THREAD

**Issue by cheako** [etemesi254/zune-image#227]
I used bisect to locate this commit https://github.com/image-rs/image/commit/a3da32ba1170d54bba9a59ac7f2cc87784b40a93 as responsible for this issue https://github.com/image-rs/image/issues/2310

I find it alarming that one crate can break another, but even more alarming that this is not something treated with more curiosity.

**Comment by etemesi254** [comment]
We had a lib type dylib that may have caused errors.

The solution may be to update to the new version

**Comment by cheako** [comment]
Thanks, I *bisected to this commit 242e89fdf7d82f8bd41dcdde5ca0678d11d8fe81 because the commit one previous(2_42e89fdf7d82f8bd41dcdde5ca0678d11d8fe81^) had the linking errors and this commit breaks the API.  I would need help from the `image` team to update that crate to continue and I can't tell if I was being unreasonable with them...  Usually when the only ppl who would have any clue refuse to acknowledge they are not perfect I end up being banned.

* The bisect was kinda strange, I thought to say good when I encountered the linking error and bad when I encountered rust type errors.  I'm certain that's what I was supposed to do, even though it being backwards to how I bisected the `image` crate.

As I've said this is simply reproduced with boilerplate code, but it's clear to me that any solution belongs in the `image` crate and not in `zune`...  as the `dev` branch of `zune` isn't being used in `image` so changes don't propegate.
